src/restic/node.go

```diff
@@ -437,45 +442,20 @@ func (node Node) sameExtendedAttributes(other Node) bool {
 	if len(node.ExtendedAttributes) != len(other.ExtendedAttributes) {
 		return false
 	}
-
-	// build a set of all attributes that node has
```

Comment:
If it is only this check that bothers you, we can also just create a `map[string][]byte` for each node on the fly and use your new code, but leave the JSON structure as it is.